enumerables.rb

```diff
@@ -0,0 +1,273 @@
+      if block.arity == 0
+        yield
+      elsif block.arity == 1
+        i = 0
+        while i < iterable_array.length
+          yield(iterable_array[i])
+          i += 1
+        end
+        return self
+      elsif block.arity >= 2
+        i = 0
+        while i < iterable_array.length
+          yield(iterable_array[i][0], iterable_array[i][1])
+          i += 1
+        end
+      end
+      self
+    end
+
+    while i <= iterable_array.length - 1
+      yield(iterable_array[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    object_to_return = []
+    if is_a? Range
+      to_a.my_each { |elt| object_to_return.push(elt) if yield(elt) }
+      return object_to_return
+    end
+    my_each do |element|
+      object_to_return << element if yield(element)
+    end
+    object_to_return
+  end
+
+  def my_all?(*arg, &block)
+    if arg.length == 1
+      i = 0
+      if arg[0].is_a? Class
+        while i < length
+          return false unless self[i].is_a? arg[0]
+
+          i += 1
+        end
+        return true
+      else
+        while i < length
+          return false if self[i] != arg
+
+          i += 1
+        end
+      end
+      return true
+    end
+    block = lambda { |obj| obj } unless block_given?
+
+    i = 0
+    while i < length
+      return false unless block.call(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?(*arg, &block)
+    if arg.length == 1
+      i = 0
+      if arg[0].is_a? Class
+        while i < length
+          return true if self[i].is_a? arg[0]
+
+          i += 1
+        end
+      else
+        i = 0
+        while i < length
+          return true if self[i] == arg
+
+          i += 1
+        end
+      end
+      return false
+    end
+    block = lambda { |obj| obj } unless block_given?
+
+    i = 0
+    while i < length
+      return true if block.call(self[i])
+
+      i += 1
+    end
+    false
+  end
+
+  def my_none?(*arg, &block)
+    !my_any?(*arg, &block)
+  end
+
+  def my_count(*arg, &block)
+    count = 0
+    if is_a? Range
+      iterable_array = to_a
+    else
+      iterable_array = self
+    end
+    if arg.length == 1
+      i = 0
+      while i < iterable_array.length
+        count += 1 if iterable_array[i] == arg[0]
+        i += 1
+      end
+      return count
+    end
+    block = lambda { |obj| obj } unless block_given?
+
+    i = 0
+    while i < iterable_array.length
+      count += 1 if block.call(iterable_array[i])
+      i += 1
+    end
+    count
+  end
+
+  def my_map
+    res = []
+    if is_a? Range
+      to_a.my_each { |elt| res.push(yield(elt)) }
+      res
+    end
+    return unless is_a? Array
+
+    return to_enum unless block_given?
+
+    my_each { |elt| res.push(yield(elt)) }
+    res
+  end
```

Comment:
## #my_map:

```ruby
  HIGHEST_VALUE = 9 
  LOWEST_VALUE = 0
  block = proc { |num| num < (LOWEST_VALUE + HIGHEST_VALUE) / 2 }
  my_proc = proc { |num| num > 10 }
```

- [ ] returns a new array with the results of running a given block once for every element in enum::range

```ruby

range.my_map(&block) === range.map(&block) => false
```

- [ ] executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.

```ruby

  array.my_map(my_proc) { |num| num < 10 } === array.map(&my_proc) => false
```